server/src/test/java/org/opensearch/common/time/JavaDateMathParserTests.java

```diff
@@ -131,6 +131,11 @@ public void testBasicDates() {
         assertDateMathEquals("2014-05-30T20:21:35.123", "2014-05-30T20:21:35.123");
     }
 
+    public void testDayOfYearWithMissingFields() {
+        DateFormatter formatter = DateFormatter.forPattern("yyyy[-DDD'T'HH:mm:ss.SSS]");
+        assertDateMathEquals(formatter.toDateMathParser(), "2022", "2022-01-01T23:59:59.999Z", 0, true, ZoneOffset.UTC);
+    }
+
```

Comment:
Not sure if it is necessary. Maybe add more test with parse date in yyyy-DDD format can be better to test this bug?

Reply:
added unit test